src/main/java/flaxbeard/immersivepetroleum/common/blocks/metal/TileEntityPumpjack.java

```diff
@@ -146,6 +147,31 @@ public void update()
 		update(true);
 	}
 
+	// Checks whether or not this pump has a pipe to bedrock.
+	private boolean hasPipe() {
+		if (!IPConfig.Extraction.require_pipes) {
+			return true;
+		}
+		
+		World world = this.getWorld();
+		BlockPos pos = this.getOrigin();
+		int depthCount = 0;
+		int depthBounds = 256;
+		while (depthCount < depthBounds) {
```

Comment:
Nice catch, there used to be another conditional in the while loop but once I removed it this is just a less-clear for loop.